src/Utilities/FlowAnalysis/Analysis/PropertySetAnalysis/PropertySetAnalysis.cs

```diff
@@ -0,0 +1,90 @@
+            ControlFlowGraph cfg,
+            Compilation compilation,
+            ISymbol owningSymbol,
+            string typeToTrackMetadataName,
+            bool isNewInstanceFlagged,
+            string propertyToSetFlag,
+            bool isNullPropertyFlagged,
+            ImmutableHashSet<string> methodNamesToCheckForFlaggedUsage,
+            InterproceduralAnalysisKind interproceduralAnalysisKind = InterproceduralAnalysisKind.ContextSensitive,
+            bool pessimisticAnalysis = false)
+        {
+            var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(compilation);
+            var pointsToAnalysisResult = PointsToAnalysis.GetOrComputeResult(
+                cfg, owningSymbol, wellKnownTypeProvider, interproceduralAnalysisKind, pessimisticAnalysis);
+
+            var analysisContext = PropertySetAnalysisContext.Create(PropertySetAbstractValueDomain.Default,
+                wellKnownTypeProvider, cfg, owningSymbol, interproceduralAnalysisKind, pessimisticAnalysis, pointsToAnalysisResult, GetOrComputeResultForAnalysisContext,
+                typeToTrackMetadataName,
+                isNewInstanceFlagged,
+                propertyToSetFlag,
+                isNullPropertyFlagged,
+                methodNamesToCheckForFlaggedUsage);
+            var result = GetOrComputeResultForAnalysisContext(analysisContext);
+            return result.HazardousUsages;
+        }
+
+        private static PropertySetAnalysisResult GetOrComputeResultForAnalysisContext(PropertySetAnalysisContext analysisContext)
+        {
+            var operationVisitor = new PropertySetDataFlowOperationVisitor(analysisContext);
+            var analysis = new PropertySetAnalysis(BinaryFormatterAnalysisDomainInstance, operationVisitor);
+            return analysis.GetOrComputeResultCore(analysisContext, cacheResult: true);
+        }
+
+        internal override PropertySetAnalysisResult ToResult(
+            PropertySetAnalysisContext analysisContext,
+            DataFlowAnalysisResult<PropertySetBlockAnalysisResult, PropertySetAbstractValue> dataFlowAnalysisResult)
+        {
+            //analysisContext = analysisContext.WithTrackHazardousParameterUsages();
+            //var newOperationVisitor = new BinaryFormatterDataFlowOperationVisitor(analysisContext);
+            //var resultBuilder = new DataFlowAnalysisResultBuilder<BinaryFormatterAnalysisData>();
+            //foreach (var block in analysisContext.ControlFlowGraph.Blocks)
+            //{
+            //    var data = BinaryFormatterAnalysisDomainInstance.Clone(dataFlowAnalysisResult[block].InputData);
+            //    data = Flow(newOperationVisitor, block, data);
+            //}
+
+            //return new BinaryFormatterAnalysisResult(dataFlowAnalysisResult, newOperationVisitor.HazardousUsages);
+
+            // Hey Manish, is it okay to just look at this.OperationVisitor?
```

Comment:
Does this make sense?

Reply:
The reason I did this in two passes for the other analyzer was that computing hazardous usages in first pass, when the analysis data has still not reached fixed point, could lead to us marking a non-hazardous usage as a hazardous one. For example, assume a usage at a start of the loop. If we haven't yet analyzed the back edge, we might mark it hazardous and find that the use from the back edge is non-hazardous. If the analysis wants to be conservative and only report hazardous usages if it is hazardous on all control flow paths, then it might lead to incorrect report. If your analysis is instead aggressive and wants to report a potentially hazardous usage on any control flow path, then your current approach should be fine.

---
In reply to: [222068138](https://github.com/dotnet/roslyn-analyzers/pull/1838#discussion_r222068138) [](ancestors = 222068138)

Reply:
For this case, I think I want more aggressive analysis.  I was having trouble getting the two pass approach working, but I'd still like to understand what I was doing wrong, so I'll sync up with you later.